consensus/dummy/dynamic_fees.go

```diff
@@ -37,22 +37,37 @@ var (
 
 // CalcExtraPrefix takes the previous header and the timestamp of its child
 // block and calculates the expected extra prefix for the child block.
-//
-// CalcExtraPrefix should only be called if [timestamp] >= [config.ApricotPhase3Timestamp]
 func CalcExtraPrefix(
 	config *params.ChainConfig,
 	parent *types.Header,
 	timestamp uint64,
 ) ([]byte, error) {
-	window, err := calcFeeWindow(config, parent, timestamp)
-	return window.Bytes(), err
+	switch {
+	case config.IsApricotPhase3(timestamp):
+		window, err := calcFeeWindow(config, parent, timestamp)
+		return window.Bytes(), err
+	default:
+		// Prior to AP3 there was no expected extra prefix.
+		return nil, nil
+	}
 }
 
 // CalcBaseFee takes the previous header and the timestamp of its child block
 // and calculates the expected base fee for the child block.
 //
-// CalcBaseFee should only be called if [timestamp] >= [config.ApricotPhase3Timestamp]
+// Prior to AP3, the returned base fee will be nil.
 func CalcBaseFee(config *params.ChainConfig, parent *types.Header, timestamp uint64) (*big.Int, error) {
+	switch {
+	case config.IsApricotPhase3(timestamp):
+		return calcBaseFeeWithWindow(config, parent, timestamp)
+	default:
+		// Prior to AP3 the expected base fee is nil.
+		return nil, nil
+	}
+}
+
+// calcBaseFeeWithWindow should only be called if [timestamp] >= [config.ApricotPhase3Timestamp]
```

Comment:
I believe using brackets won't work on argument variables. You could use `[params.ChainConfig.ApricotPhase3Timestamp]` otherwise, but that's a bit confusing. I would recommend sticking on backticks, such that the dev doesn't try to ctrl+click the non-working link

```suggestion
// calcBaseFeeWithWindow should only be called if `timestamp` >= `config.ApricotPhase3Timestamp`
```